Sporadic build failure StreamMetadataTasksTest > updateStreamTest

**Problem description**
io.pravega.controller.task.Stream.StreamMetadataTasksTest > updateStreamTest FAILED
    java.lang.AssertionError: expected:<SUCCESS> but was:<FAILURE>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at io.pravega.controller.task.Stream.StreamMetadataTasksTest.updateStreamTest(StreamMetadataTasksTest.java:222)


**Problem location**
StreamMetadataTasksTest 
It calls two update stream tasks concurrently to make sure first one passes and the second fails.. 
Since its all asynchronous processing, so sometimes second one is scheduled first before first and first one fails. 

**Suggestions for an improvement**
ensure first one starts before second one is scheduled such that the test can verify in a guaranteed manner.

--- a/controller/src/test/java/io/pravega/controller/task/Stream/StreamMetadataTasksTest.java
+++ b/controller/src/test/java/io/pravega/controller/task/Stream/StreamMetadataTasksTest.java
@@ -213,10 +213,7 @@ public class StreamMetadataTasksTest {
         AtomicReference<State> streamState = new AtomicReference<>(State.ACTIVE);
         FutureHelpers.loop(() -> streamState.get().equals(State.ACTIVE),
                 () -> streamStorePartialMock.getState(SCOPE, stream1, true, null, executor)
-                        .thenAccept(state -> {
-                            streamState.set(state);
-
-                        }), executor).join();
+                        .thenAccept(streamState::set), executor).join();
 
         StreamConfiguration streamConfiguration2 = StreamConfiguration.builder()
                 .scope(SCOPE)

--- a/controller/src/test/java/io/pravega/controller/task/Stream/StreamMetadataTasksTest.java
+++ b/controller/src/test/java/io/pravega/controller/task/Stream/StreamMetadataTasksTest.java
@@ -72,6 +72,7 @@ import java.util.concurrent.CompletionException;
 import java.util.concurrent.Executors;
 import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicReference;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
@@ -206,6 +207,17 @@ public class StreamMetadataTasksTest {
 
         CompletableFuture<UpdateStreamStatus.Status> updateOperationFuture1 = streamMetadataTasks.updateStream(SCOPE, stream1,
                 streamConfiguration1, null);
+
+        // ensure that previous updatestream has posted the event and set status to updating,
+        // only then call second updateStream
+        AtomicReference<State> streamState = new AtomicReference<>(State.ACTIVE);
+        FutureHelpers.loop(() -> streamState.get().equals(State.ACTIVE),
+                () -> streamStorePartialMock.getState(SCOPE, stream1, true, null, executor)
+                        .thenAccept(state -> {
+                            streamState.set(state);
+
+                        }), executor).join();
+
 
         StreamConfiguration streamConfiguration2 = StreamConfiguration.builder()
                 .scope(SCOPE)

--- a/controller/src/test/java/io/pravega/controller/task/Stream/StreamMetadataTasksTest.java
+++ b/controller/src/test/java/io/pravega/controller/task/Stream/StreamMetadataTasksTest.java
@@ -218,7 +218,6 @@ public class StreamMetadataTasksTest {
 
                         }), executor).join();
 
-
         StreamConfiguration streamConfiguration2 = StreamConfiguration.builder()
                 .scope(SCOPE)
                 .streamName(stream1)

